tracer/propagation_ot.go

```diff
@@ -58,14 +78,46 @@ func (p *textMapPropagator) Extract(
 		return nil, opentracing.ErrInvalidCarrier
 	}
 	requiredFieldCount := 0
-	var traceID, spanID uint64
+	var traceID uuid.UUID
+	var spanID uint64
 	var sampled bool
 	var err error
 	decodedBaggage := make(map[string]string)
+
+	err = carrier.ForeachKey(func(k, v string) error {
+		switch strings.ToLower(k) {
+		case traceParentKey:
+			if len(v) < 55 {
+				return opentracing.ErrSpanContextCorrupted
+			}
+			traceParentArray := strings.Split(v, "-")
+			if len(traceParentArray) < 4 || traceParentArray[0] != "00" || len(traceParentArray[1]) != 32 || len(traceParentArray[2]) != 16 {
+				return opentracing.ErrSpanContextCorrupted
+			}
+
+			traceID, err = uuid.Parse(traceParentArray[1])
```

Comment:
In the `Inject` method, it is being removed the `-` characters, and here it is being parse without them. Can`uuid.Parse` function parse UUIDs without `-`?